java/org/apache/catalina/tribes/membership/cloud/DNSMembershipProvider.java

```diff
@@ -30,6 +30,61 @@
 import org.apache.juli.logging.Log;
 import org.apache.juli.logging.LogFactory;
 
+/**
+ * A {@link org.apache.catalina.tribes.MembershipProvider} that uses DNS to retrieve the members of a cluster.<br>
+ *
+ * <p>
+ * <strong>Configuration example for Kubernetes</strong>
+ * </p>
+ *
+ * {@code server.xml }
+ *
+ * <pre>
+ * {@code
+ * <Server ...
+ *
+ *   <Service ...
+ *
+ *     <Engine ...
+ *
+ *       <Host ...
+ *
+ *         <Cluster className="org.apache.catalina.ha.tcp.SimpleTcpCluster">
+ *           <Channel className="org.apache.catalina.tribes.group.GroupChannel">
+ *             <Membership className="org.apache.catalina.tribes.membership.cloud.CloudMembershipService"
+ *                 membershipProviderClassName="org.apache.catalina.tribes.membership.cloud.DNSMembershipProvider"/>
+ *           </Channel>
+ *         </Cluster>
+ *         ...
+ *  }
+ *  </pre>
+ *
+ * {@code dns-membership-service.yml }
+ *
+ * <pre>
+ * {@code
+ * apiVersion: v1
```

Comment:
This is Kubernetes config. Why it is in DNSMembershipProvider ?

Reply:
At line 150 it uses Kubernetes logging message too.

Reply:
One uses rhe Kubernetes API and parses the JSON, the other one is "lightweight" and does the same with a DNS hack request. Personally, I have found the DNS technique wasn't so reliable, so I don't like it. But this is "recommended" as the proper way (access to the REST API) needs security tuning.
At line 150, it uses Kubernetes logging message because the two do the same thing and run into the same error. If you would want a separate key "because", feel free to "fix" it.

Reply:
I am confused. 
`"One uses rhe Kubernetes API..."` - is this DNSMembershipProvider ?
`...the other one is "lightweight" and does the same with a DNS hack request` - is this KubernetesMembershipProvider ?
If this is the case then IMO those classes should swap their names.

Reply:
DNSMembershipProvider does the DNS hack request, and KubernetesMembershipProvider is the one using the Kubernetes REST API.